### DISCUSSION TITLE
Incorporating edge features in a custom message passing module

### DISCUSSION THREAD

**Discussion by Angelo92Git** [pyg-team/pytorch_geometric#9130] [Q&A] [answered]
How do we incorporate edge attributes in a custom message passing module? The tutorial doesn't address this scenario.
https://pytorch-geometric.readthedocs.io/en/latest/tutorial/create_gnn.html

**Comment by rusty1s** [comment, marked as answer]
If you pass them to `propagate`, you can directly use them within `message` for example:
```python
    self.propagate(edge_index, x=x=, edge_attr=edge_attr)

def message(self, x_i, x_j, edge_attr):
    return torch.cat([x_i, x_j, edge_attr], dim=-1)
